### ISSUE TITLE
The `-single-line` and `-multi-line` options of the two `declaration-colon-*-after` rules read the run in front of a closing brace as a line of a custom property that swallowed it

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#689]
### What minimal example or steps are needed to reproduce the bug?

A custom property keeps its value's trailing whitespace, and where no semicolon closes the declaration it keeps whatever stands in front of the closing brace as well — the same reading [#538](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/538) is about, on a different pair of rules. So the break in front of the brace is inside the value, and the two `declaration-colon-*-after` rules count it as a line of the declaration.

```css
a {
	--x: pink
}
```

`declaration-colon-newline-after: "always-multi-line"` reports `2:5 Expected newline after ":" with a multi-line declaration` here and `--fix` writes the break in, leaving the value on a line of its own:

```css
a {
	--x:
 pink
}
```

Nothing in that file is multi-line: the declaration is `--x: pink`, one line, and the break belongs to the block. The same file with a semicolon behind the value draws nothing, and so does the same block with an ordinary property, since neither swallows the run.

The other face is a false negative on the sibling rule. `declaration-colon-space-after: "always-single-line"` passes over

```css
a {
	--x:pink
}
```

in silence, where it reports `2:6` on the same declaration with a semicolon behind it and `2:8` on the same block written with `color`.

Both faces reach plain CSS, `postcss-scss` and `postcss-less` alike, measured under each of the three namespaces.

Reached in one run of `--fix` as well, without the shape being written by hand: `declaration-block-trailing-semicolon: "never"` takes the semicolon off `a {⏎\t--x: pink;⏎}⏎` and the file `--fix` leaves is the one above, so the run after writes the break in and the two runs disagree. Over a matrix of 130 texts, ten neighbour settings and both listing orders, four of those 2 600 runs have a second `--fix` that differs from the first, and all four are this configuration — the same declaration in a rule and in a nested one, under either listing order.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/declaration-colon-newline-after": "always-multi-line"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

Whether a declaration is single-line or multi-line is read from the declaration, so a custom property spelling `--x: pink` is single-line whatever stands between it and the closing brace.

### What actually happened?

The run in front of the closing brace is counted as a line of the value wherever no semicolon closes the declaration, so `always-multi-line` writes a break the file never asked for and `always-single-line` passes over a declaration it should report.

### Do you have a proposal to fix the bug?

The plugin already holds the reading this wants. `declarationValueAsSpelled` takes a custom property's trailing run as the run in front of the semicolon ([#487](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/487)), and `valueAsClosed` of `lib/utils/closedBySemicolon/` reads that run as `declaration-block-trailing-semicolon` will leave it ([#536](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/536)); neither covers the declaration that carries no semicolon at all, where the run is the block's and not the value's. One reading of where a custom property's value ends would answer both rules and both faces.

Found while working on issue #358, where the pair matrix above was measured.
